R/dbi-result.R

```diff
@@ -139,6 +139,24 @@ setMethod("dbBind", "OdbcResult",
       return(invisible(res))
     }
 
+    paramDfs <- sapply(params, is.data.frame)
+    if (any(paramDfs)) {
+      # When binding table-valued-parameters verify that:
+      # * The `batch_rows` parameter is either unset or equal to one.
+      # * All non-tvp parameters are of length 1.
+      if (!is.na(batch_rows) && batch_rows > 1) {
+        cli::cli_warn("When passing data.frame(s) as query parameters,
+                      the `batch_rows` parameter must be set to 1")
+      }
+      batch_rows <- 1
+      paramLengths <- sapply(params[!paramDfs], length)
+      if (any(paramLengths > 1))
+      {
+        cli::cli_abort("When mixing data.frame(s) with other parameter types,
+                       all non-df parameters must be of length one")
+      }
+    }
+
     if (is.na(batch_rows)) {
       batch_rows <- length(params[[1]])
     }
```

Comment:
I believe if the first parameter is a data frame here, `batch_rows` will be overwritten to the number of columns of the data frame rather than 1?